File: Java/1056. Confusing Number.java
class Solution {
    public boolean confusingNumber(int n) {
        Map<Integer, Integer> mapping = new HashMap();

        mapping.put(0, 0);
        mapping.put(1, 1);
        mapping.put(6, 9);
        mapping.put(8, 8);
        mapping.put(9, 6);

        int num = n, ans = 0, rem;
        while (num > 0) {
            rem = num % 10;
            if (!mapping.containsKey(rem)) {
                return false;
            }
            ans = ans * 10 + mapping.get(rem);
            num /= 10;
        }

        return ans != n;
    }
}

File: Python3/1056. Confusing Number.py
class Solution:
    def confusingNumber(self, n: int) -> bool:
        mapping = {0: 0,
                   1: 1,
                   6: 9,
                   8: 8,
                   9: 6}

        ans = ''
        for digit in str(n):
            d = int(digit)
            if d not in mapping:
                return False
            ans += str(mapping[d])
        
        return int(ans[::-1]) != n